src/common/blkdev.cc

```diff
@@ -357,6 +373,73 @@ bool get_vdo_utilization(int fd, uint64_t *total, uint64_t *avail)
   return true;
 }
 
+// trying to use udev first, and if it doesn't work, we fall back to 
+// reading /sys/block/$devname/device/(vendor/model/serial).
+std::string get_device_id(const std::string& devname)
+{
+  struct udev_device *dev;
+  static struct udev *udev;
+  const char *data;
+  std::string device_id;
+
+  udev = udev_new();
+  if (!udev) {
+    //derr << "failed to run udev_new(), when calling for device " << devname << dendl;
+    return {};
+  }
+  dev = udev_device_new_from_subsystem_sysname(udev, "block", devname.c_str());
+  if (!dev) {
+    //derr << "failed to run udev_device_new_from_subsystem_sysname() for " << devname << dendl;
+    udev_unref(udev);
+    return {};
+  }
+
+  // "ID_SERIAL_SHORT" returns only the serial number;
+  // "ID_SERIAL" returns vendor model_serial.
+  data = udev_device_get_property_value(dev, "ID_SERIAL");
+  device_id = data;
```

Comment:
should check for failure.

see http://pulpito.ceph.com/kchai-2018-06-09_05:16:29-rados-wip-kefu-testing-2018-06-09-1015-distro-basic-smithi/

Reply:
fixed.  i'll retest in my next batch